imports/redolog_manager_factory.go

```diff
@@ -0,0 +1,157 @@
+//  Copyright (c) 2017-2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imports
+
+import (
+	"fmt"
+	"github.com/Shopify/sarama"
+	"github.com/uber/aresdb/common"
+	"github.com/uber/aresdb/diskstore"
+	memCom "github.com/uber/aresdb/memstore/common"
+	"github.com/uber/aresdb/metastore"
+	metaCom "github.com/uber/aresdb/metastore/common"
+	"github.com/uber/aresdb/utils"
+	"sync"
+)
+
+// Factory class to create shard level redolog manager
+type RedologManagerFactory struct {
+	sync.Mutex
+	namespace string
+	// flag to enable/disable local redolog
+	enableLocalRedoLog bool
+	// kafka consuer if kafka consumer is configured
+	consumer sarama.Consumer
+	// DiskStore
+	diskStore diskstore.DiskStore
+	// Metastore
+	metaStore metastore.MetaStore
+	// save all table partition redolog managers
+	managers map[string]map[int]RedologManager
+}
+
+// Function definitio to store upsertbatch
+type BatchStoreFunc func(batch *memCom.UpsertBatch, redoFile int64, offset uint32, skipBackFillRows bool) error
+type newKafkaConsumerFunc func([]string) (sarama.Consumer, error)
+
+// NewRedologManagerFactory create ImportsFactory instance
+func NewRedologManagerFactory(c *common.ImportsConfig, diskStore diskstore.DiskStore, metaStore metastore.MetaStore) (*RedologManagerFactory, error) {
+	return NewKafkaRedologManagerFactory(c, diskStore, metaStore, nil)
+}
+
+func NewKafkaRedologManagerFactory(c *common.ImportsConfig, diskStore diskstore.DiskStore, metaStore metastore.MetaStore, consumer sarama.Consumer) (*RedologManagerFactory, error) {
+
+	if c != nil && c.Source == common.KafkaSoureOnly && len(c.KafkaConfig.Brokers) == 0 {
+		return nil, fmt.Errorf("No kafka broker info configured")
```

Comment:
To more specific about the error, "No kafka broker configured for KafkaSoureOnly"